Model/Api/OAuthRedirect.php

```diff
@@ -259,6 +296,37 @@ public function login($code = '', $scope = '', $state = '', $order_id = '')
         }
     }
 
+    /**
+     * Update all immutable (child) quotes with logged-in customer
+     *
+     * @param Quote               $parentQuote
+     * @param CustomerInterface   $customer
+     */
+    private function updateImmutableQuotes($parentQuote, $customer)
+    {
+        $quoteTable = $this->resourceConnection->getTableName('quote');
+        $connection = $this->resourceConnection->getConnection();
+        $select = $connection->select()
+            ->from(
+                ['c' => $quoteTable],
+                ['*']
+            )
+            ->where('c.bolt_parent_quote_id = :bolt_parent_quote_id')
+            ->where('c.entity_id != :entity_id');
+        $bind = [
+            'bolt_parent_quote_id' => $parentQuote->getId(),
+            'entity_id' => $parentQuote->getId()
+        ];
+
+        $results = $connection->fetchAll($select, $bind);
+        foreach ($results as $data) {
+            $immutableQuote = $this->quoteFactory->create();
+            $immutableQuote->setData($data);
+            $immutableQuote->setCustomer($customer);
+            $this->cartHelper->saveQuote($immutableQuote);
+        }
```

Comment:
@nickfiacco-bolt apologies for athe late review. Writing it anyway, hopefully you find it helpful.

Although this approach is faster, we should probably use Collections instead, following Magento principles and practices.

Inject \Magento\Quote\Model\ResourceModel\Quote\CollectionFactory in the constructor, create an instance using create, addFieldToFilter('bolt_parent_quote_id', $parentQuote->getId()) and addFieldToFilter('entity_id', ['neq'=>$parentQuote->getId()]). Then foreach on the Collection and make the required changes.

Reply:
I saw something suggesting that online. I'll work on that so it's ready before 2.22. Thanks for the advice!